fmlloader/src/main/java/net/minecraftforge/fml/common/asm/ObjectHolderDefinalize.java

```diff
@@ -109,4 +120,5 @@ public int processClassWithFlags(final Phase phase, final ClassNode classNode, f
         return changes.get() ? ComputeFlags.SIMPLE_REWRITE : ComputeFlags.NO_REWRITE;
     }
 
+    public record VanillaObjectHolderData(String holderClass, String registryName, String registryType) {}
```

Comment:
It may be worth just duplicating this in `ObjectHolderRegistry` and marking this class as `private`.